subprojects/testing-base/src/integTest/groovy/org/gradle/testing/TestListenerBuildOperationAdapterIntegrationTest.groovy

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.api.tasks.testing.ExecuteTestBuildOperationType
+import org.gradle.api.tasks.testing.TestOutputBuildOperationType
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.integtests.fixtures.TestResources
+import org.gradle.internal.operations.trace.BuildOperationRecord
+import org.junit.Rule
+
+class TestListenerBuildOperationAdapterIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
This class is poorly named as it leaks unnecessary implementation specifics. Should be `TestExecutionBuildOperationsIntegrationTest` or something similar.

Reply:
fixed